# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/shell/js.cpp
# Commit: 11fc8f557a2a
# Full Hash: 11fc8f557a2a7cf3c89e7c1396316aabd9725778
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 9bfa6023a553 -r 11fc8f557a2a js/src/shell/js.cpp
--- a/js/src/shell/js.cpp	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/shell/js.cpp	Sat Mar 06 01:05:21 2021 +0000
@@ -4653,21 +4653,6 @@
   return true;
 }
 
-static bool UnwrappedObjectsHaveSameShape(JSContext* cx, unsigned argc,
-                                          JS::Value* vp) {
-  CallArgs args = CallArgsFromVp(argc, vp);
-  if (!args.get(0).isObject() || !args.get(1).isObject()) {
-    JS_ReportErrorASCII(cx, "2 objects expected");
-    return false;
-  }
-
-  RootedObject obj1(cx, UncheckedUnwrap(&args[0].toObject()));
-  RootedObject obj2(cx, UncheckedUnwrap(&args[1].toObject()));
-
-  args.rval().setBoolean(obj1->shape() == obj2->shape());
-  return true;
-}
-
 static bool Sleep_fn(JSContext* cx, unsigned argc, Value* vp) {
   ShellContext* sc = GetShellContext(cx);
   CallArgs args = CallArgsFromVp(argc, vp);
@@ -9045,12 +9030,6 @@
 "groupOf(obj)",
 "  Get the group of obj (an implementation detail)."),
 
-    JS_FN_HELP("unwrappedObjectsHaveSameShape", UnwrappedObjectsHaveSameShape, 2, 0,
-"unwrappedObjectsHaveSameShape(obj1, obj2)",
-"  Returns true iff obj1 and obj2 have the same shape, false otherwise. Both\n"
-"  objects are unwrapped first, so this can be used on objects from different\n"
-"  globals."),
-
 #ifdef DEBUG
     JS_FN_HELP("arrayInfo", ArrayInfo, 1, 0,
 "arrayInfo(a1, a2, ...)",